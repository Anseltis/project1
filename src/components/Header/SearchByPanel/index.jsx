import React from 'react'

import './style.scss'

export const SearchByPanel = ({className, searchBy}) => {
    return (
        <div className={className} onClick={searchBy}>
        <span className={`${className}__header`}>Search by</span>
            <input className={`${className}__title-input`} type={'radio'} name={'search-by'} value={'title'} id={'title'} defaultChecked={true}/>
            <label className={`${className}__title-label label-chosen`} htmlFor={'title'}>Title</label>
            <input className={`${className}__genre-input`} type={'radio'} name={'search-by'} value={'genre'} id={'genre'}/>
            <label className={`${className}__genre-label`} htmlFor={'genre'}>Genre</label>
        </div>
    )
}
