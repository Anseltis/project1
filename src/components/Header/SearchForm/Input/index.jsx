import React from 'react'
import './style.scss'

export const Input = ({className, placeholder, onChange, value}) => {
    return (
        <input className={className} type="text" placeholder={placeholder} onChange={onChange} value = {value}/>
    )
}
