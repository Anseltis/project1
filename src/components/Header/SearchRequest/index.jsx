import React from 'react'

import { SearchForm } from 'components/form/index'

const createXHR = (onLoad) => {
    const xhr = new window.XMLHttpRequest()
    xhr.responseType = 'json'
    xhr.timeout = 10000

    xhr.addEventListener('load', function () {
        switch (xhr.status) {
            case 200:
                onLoad(xhr.response)
                break
        }
    })
    return xhr

const load = (onLoad) => {
    const xhr = createXHR(onLoad)
    const request = 'pulp fiction'
    const by = '&searchBy=title'
    const URL = request + by
    xhr.open('GET', URL)
    }
}
class SearchRequest extends React.Component {

}

