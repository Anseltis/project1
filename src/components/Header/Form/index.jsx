import React from 'react'

import { Input } from 'components/Header/Form/Input'
import { SearchBy } from 'components/Header/SearchBy'
import './style.scss'

export const SearchForm = ({className, onChange, onSubmit, value}) => {
    return (
        <form className={`${className} main-search`} onSubmit={onSubmit}>
            <h1 className={'main-search__h1'}>Find your movie</h1>
            <Input className = {'main-search__input'} placeholder={'Quentin Tarantino'} onChange={onChange} value = {value}/>
            <div className={'main-search__container'}>
            <SearchBy className={'search-by'}/>
            <input type="submit" className={'main-search__button'} />
            </div>
        </form>
    )
}
