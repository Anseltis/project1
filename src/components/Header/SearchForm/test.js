import React from 'react'

import { SearchForm } from 'components/Header/SearchForm/index'
import renderer from 'react-test-renderer'

test('renders Input properly', () => {
    const wrapper = renderer
        .create(<SearchForm/>)
        .toJSON()
    expect(wrapper).toMatchSnapshot()

})