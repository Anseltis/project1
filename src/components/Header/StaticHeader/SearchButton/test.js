import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { SearchForm } from 'components/Header/SearchForm'
import { SearchButton } from './index'

describe('SearchButton', ()=> {
    test('SearchButton work correct with received props', () => {
        const wrapper = shallow(<SearchButton classAttribute="testing"/>)
        expect(wrapper.find('button').hasClass('testing')).toEqual(true)
    })
    test('renders SearchButton properly', ()=> {
        const wrapper = renderer
            .create(<SearchForm/>)
            .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
})

