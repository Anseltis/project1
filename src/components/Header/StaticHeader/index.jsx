import React from 'react'

import { MainLogo } from '../../MainLogo'
import { SearchButton } from './SearchButton'

import './style.scss'

export const StaticHeader = ({ className, onclick }) => {
  return (
    <nav className={`header__static ${className}`}>
      <MainLogo className="navigation__logo" />
      <SearchButton
        classAttribute="navigation__search"
        value="search"
        action={onclick}
      />
    </nav>
  )
}
