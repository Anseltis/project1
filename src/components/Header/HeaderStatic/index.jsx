import React from 'react'

import { MainLogo } from 'components/MainLogo'
import { SearchButton } from 'components/Header/HeaderStatic/SearchButton'
import './style.scss'

export const HeaderStatic = ({className}) => {
    return (
        <nav className= {`header__static ${className}`} >
            <MainLogo className={'navigation__logo'}/>
            <SearchButton classAttribute='navigation__search' value='search'/>
        </nav>
    )
}
