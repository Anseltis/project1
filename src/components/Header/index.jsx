import React from 'react'

import { HeaderStatic } from 'components/Header/HeaderStatic'
import { SearchForm } from 'components/Header/Form'
import { MovieCard } from 'components/MovieCard'
import { filmsArray } from '../../../assets/data'
import './style.scss'

const film = filmsArray[0]

export const Header = ({className}) => {
    return <header className = {`header ${className}`} >
        <HeaderStatic className = {'navigation'}/>
        <MovieCard film={film}/>
        <SearchForm className = "header__form"/>
    </header>
}
