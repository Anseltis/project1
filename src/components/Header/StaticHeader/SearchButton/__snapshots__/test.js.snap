// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton renders SearchButton properly 1`] = `
<form
  className="undefined main-search"
>
  <h1
    className="main-search__h1"
  >
    Find your movie
  </h1>
  <input
    className="main-search__input"
    placeholder="Quentin Tarantino"
    type="text"
  />
  <div
    className="main-search__container"
  >
    <div
      className="search-by"
    >
      <span
        className="search-by__header"
      >
        Search by
      </span>
      <input
        className="search-by__title-input"
        defaultChecked={false}
        id="title"
        name="search-by"
        type="radio"
        value="title"
      />
      <label
        className="search-by__title-label"
        htmlFor="title"
      >
        Title
      </label>
      <input
        className="search-by__genre-input"
        defaultChecked={false}
        id="genre"
        name="search-by"
        type="radio"
        value="genre"
      />
      <label
        className="search-by__title-label"
        htmlFor="genre"
      >
        Genre
      </label>
    </div>
    <input
      className="main-search__button"
      type="submit"
    />
  </div>
</form>
`;
